src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -26,11 +33,71 @@
     private final ItemModel model;
     private final Storage storage;
     private final AddressBookParser addressBookParser;
+    private final ScheduledThreadPoolExecutor checker;
 
     public LogicManager(ItemModel model, Storage storage) {
         this.storage = storage;
         this.model = model;
+
         addressBookParser = new AddressBookParser(model.getElisaStateHistory());
+
+        //Create new thread class to check
+        /*
+        class CheckerThread extends Thread {
+
+            ScheduledExecutorService
+            //Schedule a check on future list every 5 seconds
+            //If future.first.time <= now
+            // A new collection <- future.remove until future.first.time > now
+            // ActiveRemindersListProperty.add(A new collection)
+            //
+        }
+        */
+        //Spawn new thread
+        //new CheckerThread().start();
+        //Run checker
+        Runnable checkTask = new Runnable() {
+            public void run() {
+                System.out.println("running checkTask");
+                Reminder reminder;
+                ArrayList<Item> reminders = model.getFutureRemindersList();
+                ArrayList<Item> activeReminders = new ArrayList<Item> (0);
+                if (reminders.size() > 0) {
+                    System.out.println("There are pending reminders: " + reminders.toString());
+                    //TODO: Check if Optional is present before .get()
+                    reminder = reminders.get(0).getReminder().get();
+                    synchronized (reminders) {
+                        while (reminder != null && reminder.getDateTime().isBefore(LocalDateTime.now())) {
```

Comment:
Great work! Just wondering, how do you handle the case where the reminder is before the LocalDateTime? Does it inform the user?

Reply:
Sorry, could you clarify what you mean? The code above handles the case where the reminder is before the LocalDateTime.now(). On the off-chance you meant to ask if the reminder is after LocalDateTime.now() then it should remain in reminders list as the loop continues to check for reminders that are due to show.